pages/room/[...id].js

```diff
@@ -32,130 +33,148 @@ PeerVideo.propTypes = {
+      width: window.innerWidth / 2,
+      frameRate: { ideal: 15, max: 30 },
+    };
+
+    socketRef.current = io(process.env.NEXT_PUBLIC_NODE_SERVER || 'http://localhost:8080');
+    let stream = null;
+    try {
+      stream = await navigator.mediaDevices.getUserMedia({ video: videoConstraints, audio: true });
+      userVideo.current.srcObject = stream;
+      localStorage.setItem(USER_MEDIA_ACTIVE, true);
+    } catch (err) {
+      console.warn(err);
+    }
+
+    /**
+     * Notifiy users in the room that this new user joined
+     */
+    socketRef.current.emit('join room', { roomID, username: currentUsername });
+
+    /**
+     * Get information of all users in the room and add them as peers
+     * Also populates conversation with redis cache
+     */
+    socketRef.current.on('all users', ({ users, conversation }) => {
+      const peers = [];
+      const newParticipants = [];
+      newParticipants.push(currentUsername);
+      users.forEach((user) => {
+        const peer = createPeer(user.socketID, currentUsername, socketRef.current.id, stream);
+        peersRef.current.push({
+          peerID: user.socketID,
+          peer,
         });
-        currentUsername = randomName;
-      }
-      setUsername(currentUsername);
-      const videoConstraints = {
-        height: window.innerHeight / 2,
-        width: window.innerWidth / 2,
-      };
-
-      socketRef.current = io(process.env.NEXT_PUBLIC_NODE_SERVER || 'http://localhost:8080');
-      navigator.mediaDevices.getUserMedia({ video: videoConstraints, audio: true }).then((stream) => {
-        userVideo.current.srcObject = stream;
-
-        /**
-         * Notifiy users in the room that this new user joined
-         */
-        socketRef.current.emit('join room', { roomID, username: currentUsername });
-
-        /**
-         * Get information of all users in the room and add them as peers
-         */
-        socketRef.current.on('all users', (users) => {
-          const peers = [];
-          const newParticipants = [];
-          newParticipants.push(currentUsername);
-          users.forEach((user) => {
-            const peer = createPeer(user.socketID, currentUsername, socketRef.current.id, stream);
-            peersRef.current.push({
-              peerID: user.socketID,
-              peer,
-            });
-            peers.push(peer);
-            newParticipants.push(user.username);
-          });
-          setPeers(peers);
-          setParticipants([...newParticipants]);
+        peers.push(peer);
+        newParticipants.push(user.username);
+      });
+      setPeers(peers);
+      setParticipants([...newParticipants]);
+      if (conversation) {
+        conversation = JSON.parse(conversation).map((obj) => {
+          if (!obj) return {};
+          obj = JSON.parse(obj);
+          if (!obj?.message || !obj?.username) return {};
+          return { text: obj?.message, sender: obj?.username, fromMe: obj?.username == currentUsername };
```

Comment:
curious as to that what the structure of `conversation` is. When would `obj` be null? or not have a message and username?

Reply:
There shouldn't be but generally when communicating with the backend, it's good practice to do error checking in the off chance that something breaks. Not entirely needed tbh